features/upgrade/asb/upgrade.feature

```diff
@@ -0,0 +1,114 @@
+    # Enable Service Catalog so that the ASB can work well
+    Given enable service catalog
+    # Get the cluster version: oc get clusterversion version -o=jsonpath='{.spec.channel}' | tr -d 'stable-'
+    Given the major.minor version of the cluster is stored in the clipboard
+
+    # This step will install the OperatorGroup, operator. Please use your operator Package name in here.
+    Given optional operator "openshiftansibleservicebroker" from channel "<%= cb.operator_channel_name %>" is subscribed in "openshift-ansible-service-broker" project
+    # Check the pods of the operator
+    Then I wait for the "openshift-ansible-service-broker-operator" deployment to appear
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+      | name=openshift-ansible-service-broker-operator-alm-owned |
+    """
+    # Check if the corresponding CRD is ready
+    Then I run the :get client command with:
+      | resource | automationbroker |
+    And the output should contain "No resources found"
+    #Create Cluster Role Binding for the operator
+    When I process and create:
+      | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/asb-rolebinding-template.yaml |
+    Then the step should succeed
+    # Create customer resource for the operator
+    When I process and create:
+      | f | <%= ENV['BUSHSLICER_HOME'] %>/testdata/olm/asb-cr-template.yaml |
+    Then the step should succeed
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    And a pod becomes ready with labels:
+        | app=ansible-service-broker |
+    """
+    
+  @admin
+  @upgrade-check
+  @users=upuser1,upuser2
+  Scenario: upgrade ASB
+    Given I switch to cluster admin pseudo user
+    And I use the "openshift-ansible-service-broker" project
+    # Check if the previous ASB works works well.
+    When I run the :describe client command with:
+      | resource           | clusterservicebroker    |
+      | name               | ansible-service-broker |
+    And the output should match:
+      | Reason:\\s+FetchedCatalog |
+      | Status:\\s+True           |
+
+    # Update ASB operator to the new version
+    Given the major.minor version of the cluster is stored in the clipboard
+    When I run the :patch client command with:
+      | resource      | subscription                                              |
+      | resource_name | openshiftansibleservicebroker-sub                        |
+      | p             | {"spec": {"channel": "<%= cb.operator_channel_name %>" }} |
+      | type          | merge                                                     |
+    Then the step should succeed
+
+    And I wait up to 360 seconds for the steps to pass:
+    """
+    When I run the :get client command with:
+      | resource | clusterserviceversion |
+    Then the output should contain:
+      | <%= cb.operator_channel_name %>  |
+      | Succeeded                        |
+    """
```

Comment:
I assume this is checking cluster service cersion reported succeeded. But I guess you can see `succeeded` for some other item, not what you expect. Perhaps use step 
```
Given /^(I|admin) waits? for all (\w+) in the#{OPT_QUOTED} project to become ready(?: up to (\d+) seconds)?$/
```

Just it will be necessary to add method `ready?` to `lib/openshift/cluster_service_version.rb`.